### PR TITLE
CLOUD-735: Add preserveUnknownFields to CRD

### PR DESCRIPTION
**Pull request by egegunes**
[CLOUD-735](https://jira.percona.com/browse/CLOUD-735) [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=percona&utm_campaign=badge_info)

### CODE REVIEW

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| cross-site | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/cross-site.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/scaling.log) |
| scheduled-backup | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/scheduled-backup.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/recreate.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/storage.log) |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/upgrade-haproxy.log) |
| haproxy | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/haproxy.log) |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/big-data.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/scaling-proxysql.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/self-healing.log) |
| self-healing-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/self-healing-chaos.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/restore-to-encrypted-cluster.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/security-context.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/upgrade-proxysql.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/init-deploy.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/self-healing-advanced.log) |
| self-healing-advanced-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/self-healing-advanced-chaos.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/demand-backup.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/limits.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/smart-update.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/operator-self-healing.log) |
| operator-self-healing-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/operator-self-healing-chaos.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/monitoring-2-0.log) |
| pitr | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/pitr.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/affinity.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/one-pod.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/auto-tuning.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/proxysql-sidecar-res-limits.log) |
| demand-backup-encrypted-with-tls | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/demand-backup-encrypted-with-tls.log) |
| users | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/users.log) |
| tls-issue-self | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/tls-issue-self.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/upgrade-consistency.log) |
| tls-issue-cert-manager | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/tls-issue-cert-manager.log) |
| tls-issue-cert-manager-ref | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/tls-issue-cert-manager-ref.log) |
| validation-hook | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/validation-hook.log) |
| proxy-protocol | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/5fac4f4b/proxy-protocol.log) |
| We run 36 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1200/commits/5fac4f4b47ba7f72916e044fdb25b184ad34d364
image: `perconalab/percona-xtradb-cluster-operator:PR-1200-5fac4f4b`

File: deploy/crd.yaml [hunk 1]
```diff
@@ -3,6 +3,7 @@ kind: CustomResourceDefinition
 metadata:
   name: perconaxtradbclusters.pxc.percona.com
 spec:
+  preserveUnknownFields: false
```
Comment (tplavcic) [inline deploy/crd.yaml:6]: What are we solving with this?

**Review by tplavcic — CHANGES_REQUESTED** [review]

File: deploy/crd.yaml (hunk 1 above)
Comment (hors) [reply deploy/crd.yaml:6]: @egegunes please add task number to the commit message

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| upgrade-haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/upgrade-haproxy.log) |
| big-data | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/big-data.log) |
| haproxy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/haproxy.log) |
| scheduled-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/scheduled-backup.log) |
| scaling | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/scaling.log) |
| recreate | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/recreate.log) |
| cross-site | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/cross-site.log) |
| storage | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/storage.log) |
| scaling-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/scaling-proxysql.log) |
| self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/self-healing.log) |
| self-healing-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/self-healing-chaos.log) |
| init-deploy | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/init-deploy.log) |
| restore-to-encrypted-cluster | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/restore-to-encrypted-cluster.log) |
| security-context | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/security-context.log) |
| upgrade-proxysql | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/upgrade-proxysql.log) |
| self-healing-advanced | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/self-healing-advanced.log) |
| self-healing-advanced-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/self-healing-advanced-chaos.log) |
| demand-backup | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/demand-backup.log) |
| limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/limits.log) |
| smart-update | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/smart-update.log) |
| operator-self-healing | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/operator-self-healing.log) |
| operator-self-healing-chaos | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/operator-self-healing-chaos.log) |
| monitoring-2-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/monitoring-2-0.log) |
| affinity | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/affinity.log) |
| one-pod | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/one-pod.log) |
| pitr | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/pitr.log) |
| auto-tuning | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/auto-tuning.log) |
| proxysql-sidecar-res-limits | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/proxysql-sidecar-res-limits.log) |
| upgrade-consistency | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/upgrade-consistency.log) |
| users | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/users.log) |
| demand-backup-encrypted-with-tls | [failed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/0584f781/demand-backup-encrypted-with-tls.log) |
| We run 31 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1200/commits/0584f781e6a8d401f545481faa9f26e92e08ebbb
image: `perconalab/percona-xtradb-cluster-operator:PR-1200-0584f781`

**Review by hors — APPROVED** [review]

**Review by tplavcic — APPROVED** [review]

**Comment by JNKPercona** [comment]
| Test name  | Status |
| ------------- | ------------- |
| haproxy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/haproxy-8-0.log) |
| scheduled-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/scheduled-backup-8-0.log) |
| init-deploy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/init-deploy-8-0.log) |
| scaling-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/scaling-8-0.log) |
| upgrade-haproxy-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/upgrade-haproxy-8-0.log) |
| scaling-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/scaling-proxysql-8-0.log) |
| limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/limits-8-0.log) |
| security-context-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/security-context-8-0.log) |
| monitoring-2-0-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/monitoring-2-0-8-0.log) |
| affinity-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/affinity-8-0.log) |
| storage-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/storage-8-0.log) |
| cross-site-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/cross-site-8-0.log) |
| one-pod-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/one-pod-8-0.log) |
| self-healing-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/self-healing-8-0.log) |
| auto-tuning-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/auto-tuning-8-0.log) |
| self-healing-chaos-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/self-healing-chaos-8-0.log) |
| proxysql-sidecar-res-limits-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/proxysql-sidecar-res-limits-8-0.log) |
| self-healing-advanced-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/self-healing-advanced-8-0.log) |
| big-data-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/big-data-8-0.log) |
| recreate-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/recreate-8-0.log) |
| users-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/users-8-0.log) |
| self-healing-advanced-chaos-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/self-healing-advanced-chaos-8-0.log) |
| restore-to-encrypted-cluster-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/restore-to-encrypted-cluster-8-0.log) |
| operator-self-healing-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/operator-self-healing-8-0.log) |
| demand-backup-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/demand-backup-8-0.log) |
| operator-self-healing-chaos-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/operator-self-healing-chaos-8-0.log) |
| pitr-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/pitr-8-0.log) |
| users-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/users-5-7.log) |
| one-pod-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/one-pod-5-7.log) |
| scheduled-backup-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/scheduled-backup-5-7.log) |
| init-deploy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/init-deploy-5-7.log) |
| haproxy-5-7 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/haproxy-5-7.log) |
| upgrade-proxysql-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/upgrade-proxysql-8-0.log) |
| smart-update-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/smart-update-8-0.log) |
| upgrade-consistency-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/upgrade-consistency-8-0.log) |
| tls-issue-self-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/tls-issue-self-8-0.log) |
| tls-issue-cert-manager-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/tls-issue-cert-manager-8-0.log) |
| tls-issue-cert-manager-ref-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/tls-issue-cert-manager-ref-8-0.log) |
| demand-backup-encrypted-with-tls-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/demand-backup-encrypted-with-tls-8-0.log) |
| validation-hook-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/validation-hook-8-0.log) |
| proxy-protocol-8-0 | [passed](https://percona-jenkins-artifactory-public.s3.amazonaws.com/cloud-pxc-operator/PR-1200/9183db5a/proxy-protocol-8-0.log) |
| We run 41 out of 36|

commit: https://github.com/percona/percona-xtradb-cluster-operator/pull/1200/commits/9183db5a2e47fe3d3700a755173c212d4deb1dec
image: `perconalab/percona-xtradb-cluster-operator:PR-1200-9183db5a`

**Pull request merged**